geopandas/array.py

```diff
@@ -967,34 +967,54 @@ def take(self, indices, allow_fill=False, fill_value=None):
         return GeometryArray(result, crs=self.crs)
 
     def _fill(self, idx, value):
-        """Fill index locations with value
+        """
+        Fill index locations with ``value``.
 
-        Value should be a BaseGeometry
+        ``value`` should be a BaseGeometry or a GeometryArray.
         """
-        if not (_is_scalar_geometry(value) or value is None):
+        from pandas import Series
+
+        if vectorized.isna(value):
+            value = [BaseGeometry()]
```

Comment:
This will transform any NAN (i.e. `np.na`, `pd.NA`) value into empty polygon. This behavior is matched my experience. Or still use None like #1149.

Reply:
I just move `vectorized.isna` from `GeometryArray.fillna` to `GeometryArray._fill`.

And there will be None instead of an empty polygon if input `np.nan` now.